msteams-platform/m365-apps/actions-in-m365.md

```diff
@@ -0,0 +1,213 @@
+            "id": "addTodoTask",
+            "displayName": "Add todo task",
+            "intent": "addTo",
+            "description": "Add this file with a short note to my to do list",
+            "handlers": [
+                {
+                    "type": "openDialog",
+                    "supportedObjects": {
+                        "file": {
+                            "extensions": ["xlsx", "doc", "dot", "docx", "pdf", "pptx", "ppt"]
+                        }
+                    },
+                    "dialogInfo": {
+                        "dialogType": "url",
+                        "url": "http://localhost:53000/index.html#/dialogPage",
+                        "width": "small",
+                        "height": "large"
+                    }
+                }
+            ]
+        }
+    ]
+    ```
+
+For more information, see [public developer preview app manifest schema](../resources/schema/manifest-schema-dev-preview.md#actions).
+
+#### Create a handler
+
+Update the handler to receive the Action information through the context object to create a seamless user experience for performing users specific tasks using the [Teams JavaScript library (TeamsJS)](/javascript/api/@microsoft/teams-js).
+
+When a user selects Add option from the app's context menu, a page or dialog opens with the help of the `openDialog` property in the app manifest. Your app can access contextual information about the invoked Action from the `actionInfo` property of the context object `app.getContext()`.
+
+```javascript
+const context = await app.getContext();
+const itemId = context.actionInfo && context.actionInfo.actionObjects[0].itemId;
+this.setState({
+    itemId: itemId
+});
+```
+
+**Parameters**
+
+| &nbsp; | Name | Description |
+| --- | --- | --- |
+| &nbsp; | `actionId` | Maps to the action id supplied inside the manifest. |
+| &nbsp; | `actionObjects` | Array of corresponding action objects. |
+
+* [`actionInfo`](/javascript/api/@microsoft/teams-js/actioninfo): The TeamsJS helps to enable your app to determine when a user opens a page or dialog from an Action, and the content that initiated the Action.
+
+    ```javascript
+    app.getContext().then((context: app.Context) => {
+        const actionInfo = context.actionInfo;
+        if (actionInfo && actionInfo.actionId == 'myActionId1') {
+            // Handle specific action
+        }
+    });
+    ```
+
+* `itemId`: The app receives the id as the content and uses it to query the Microsoft graph.
+
+#### Access content through Graph API
+
+After obtaining the `itemId` of the triggering content, you can use the Graph API to read or modify the content, facilitating task completion for your users.
+
+```javascript
+async readActionItem() {
+    try {
+        return await this.graphClient.api(`/users/${this.objectId}/drive/items/${this.itemId}`).get();
+    } catch (error) {
+        console.log("readActionItem", error);
+    }
+}
+```
+
```

Comment:
We're missing the optional step of closing the dialog here (in private preview article).